src/realm/object-store/sectioned_results.cpp

```diff
@@ -254,30 +256,35 @@ ResultsSection::ResultsSection(SectionedResults* parent, Mixed key)
 
 bool ResultsSection::is_valid() const
 {
-    util::CheckedUniqueLock lock(m_parent->m_mutex);
+    return do_is_valid().first;
+}
+
+std::pair<bool, util::Optional<Section&>> ResultsSection::do_is_valid() const
```

Comment:
The bool appears to be redundant here? It never returns `{true, util::none}` or `{false, section}`, so the optional conveys all of the information by itself. Name-wise it should be something like `get_if_valid()` since it's doing more than just returning a bool.